packages/os/os.spec

```
@@ -192,19 +226,68 @@ Summary: Bottlerocket GPT priority querier/switcher
 
 %package -n %{_cross_os}updog
 Summary: Bottlerocket updater CLI
+Requires: %{_cross_os}updog(binaries)
 Conflicts: %{_cross_os}image-feature(no-in-place-updates)
 %description -n %{_cross_os}updog
 not much what's up with you
 
+%package -n %{_cross_os}updog-bin
+Summary: Bottlerocket updater CLI binaries
+Provides: %{_cross_os}updog(binaries)
+Requires: (%{_cross_os}image-feature(no-fips) and %{_cross_os}updog)
+Conflicts: (%{_cross_os}image-feature(no-in-place-updates) or %{_cross_os}image-feature(fips) or %{_cross_os}updog-fips-bin)
+%description -n %{_cross_os}updog-bin
+%{summary}.
+
+%package -n %{_cross_os}updog-fips-bin
+Summary: Bottlerocket updater CLI binaries, FIPS edition
+Provides: %{_cross_os}updog(binaries)
+Requires: (%{_cross_os}image-feature(fips) and %{_cross_os}updog)
+Conflicts: (%{_cross_os}image-feature(no-in-place-updates) or %{_cross_os}image-feature(no-fips) or %{_cross_os}updog-bin)
+%description -n %{_cross_os}updog-fips-bin
+%{summary}.
+
 %package -n %{_cross_os}metricdog
 Summary: Bottlerocket health metrics sender
+Requires: %{_cross_os}metricdog(binaries)
 %description -n %{_cross_os}metricdog
 %{summary}.
 
+%package -n %{_cross_os}metricdog-bin
+Summary: Bottlerocket health metrics sender binaries
+Provides: %{_cross_os}metricdog(binaries)
+Requires: (%{_cross_os}image-feature(no-fips) and %{_cross_os}metricdog)
+Conflicts: (%{_cross_os}image-feature(fips) or %{_cross_os}metricdog-fips-bin)
+%description -n %{_cross_os}metricdog-bin
+%{summary}.
+
+%package -n %{_cross_os}metricdog-fips-bin
+Summary: Bottlerocket health metrics sender binaries, FIPS edition
+Provides: %{_cross_os}metricdog(binaries)
+Requires: (%{_cross_os}image-feature(fips) and %{_cross_os}metricdog)
+Conflicts: (%{_cross_os}image-feature(no-fips) or %{_cross_os}metricdog-bin)
+%description -n %{_cross_os}metricdog-fips-bin
+%{summary}.
+
 %package -n %{_cross_os}logdog
 Summary: Bottlerocket log extractor
+Requires: %{_cross_os}logdog(binaries)
 %description -n %{_cross_os}logdog
-use logdog to extract logs from the Bottlerocket host
+%{summary}.
+
+%package -n %{_cross_os}logdog-bin
+Summary: Bottlerocket log extractor binaries
+Provides: %{_cross_os}logdog(binaries)
+Conflicts: (%{_cross_os}image-feature(fips) or %{_cross_os}logdog-fips-bin)
+%description -n %{_cross_os}logdog-bin
+%{summary}.
+
+%package -n %{_cross_os}logdog-fips-bin
+Summary: Bottlerocket log extractor binaries, FIPS edition
+Provides: %{_cross_os}logdog(binaries)
+Conflicts: (%{_cross_os}image-feature(no-fips) or %{_cross_os}logdog-bin)
+%description -n %{_cross_os}logdog-fips-bin
+%{summary}.
```

For overall robustness, these should require their parent `logdog` package. Otherwise a variant could request `bottlerocket-logdog-fips-bin` and not also end up with `bottlerocket-logdog`.

```suggestion
%package -n %{_cross_os}logdog-bin
Summary: Bottlerocket log extractor binaries
Provides: %{_cross_os}logdog(binaries)
Requires: (%{_cross_os}image-feature(no-fips) and %{_cross_os}logdog)
Conflicts: (%{_cross_os}image-feature(fips) or %{_cross_os}logdog-fips-bin)
%description -n %{_cross_os}logdog-bin
%{summary}.

%package -n %{_cross_os}logdog-fips-bin
Summary: Bottlerocket log extractor binaries, FIPS edition
Provides: %{_cross_os}logdog(binaries)
Requires: (%{_cross_os}image-feature(fips) and %{_cross_os}logdog)
Conflicts: (%{_cross_os}image-feature(no-fips) or %{_cross_os}logdog-bin)
%description -n %{_cross_os}logdog-fips-bin
%{summary}.
```